### ISSUE TITLE
Bug - Property named dependencies is not getting generated property

### ISSUE THREAD

**Issue by Senthil-Sivanath** [json-schema-faker/json-schema-faker#802]
If you are reporting a bug, please include the following:

1. the JSON Schema content that fails - preferably using http://json-schema-faker.js.org/ (paste the schema, click `Generate example` and then copy and paste the resulting URL from the browser address bar)

Schema - {
  "properties": {
    "dependencies": {
      "items": {
        "$ref": "Dependencies"
      }
    }
  }
}
Refs - [
  {
    "id": "Dependencies",
    "properties": {
      "someProperty": {
        "type": "string"
      }
    }
  }
]
2. the result you got
```json5
{
  "dependencies": {
    "items": {
      "properties": {
        "someProperty": "enim ad"
      }
    }
  }
}
```
3. the result you expected
```json5
{
  "dependencies1": [
    {
      "someProperty": "dolore"
    }
  ]
}
```

**Comment by pateketrueke** [comment]
You should specify `"type": "object"` in your Dependencies schema, but I am not pretty sure... try that and let me know if that works, thank you!

(this could be also true for any schema you use, we can try to guess but it's better to be more explicit)

**Comment by Senthil-Sivanath** [comment]
Got same result with `"type": "object"` as well. 

The issue is with `dependencies` name, if i use a different name it works. Internally dependencies is getting used.

**Comment by pateketrueke** [comment]
I think this is related with #801 right? If we can pre-resolve everything fine then we can jump into the resolved schema and just generate values, gonna explore that these upcoming weeks, thank you!

**Comment by Senthil-Sivanath** [comment]
Not sure if they are interconnected ? The word dependencies is creating an issue, if i use any other keyword apart from dependencies it as property name, faker generates fine.

**Comment by pateketrueke** [comment]
Yeah, our implementation is fairly weak at that point, if we encounter the "dependencies" property we treat it as a map of dependencies... which is incorrect, depending on where it's located, probably we need to rework that logic!

**Comment by Senthil-Sivanath** [comment]
Similar issue with`items` keyword as well.
